test/e2e/mcd_test.go

```diff
@@ -592,3 +599,59 @@ func TestFIPS(t *testing.T) {
 		t.Logf("Node %s has expected FIPS mode", node.Name)
 	}
 }
+
+func TestCustomPool(t *testing.T) {
+	cs := framework.NewClientSet("")
+
+	nodes, err := getNodesByRole(cs, "worker")
+	require.Nil(t, err)
+	require.NotEmpty(t, nodes)
+	infraNode := nodes[0]
+	out, err := exec.Command("oc", "label", "node", infraNode.Name, "node-role.kubernetes.io/infra=").CombinedOutput()
+	require.Nil(t, err, "unable to label worker node %s with infra: %s", infraNode.Name, string(out))
+
+	infraMCP := &mcfgv1.MachineConfigPool{}
+	infraMCP.Name = "infra"
+	nodeSelector := metav1.LabelSelector{}
+	infraMCP.Spec.NodeSelector = &nodeSelector
+	infraMCP.Spec.NodeSelector.MatchLabels = make(map[string]string)
+	infraMCP.Spec.NodeSelector.MatchLabels["node-role.kubernetes.io/infra"] = ""
+	mcSelector := metav1.LabelSelector{}
+	infraMCP.Spec.MachineConfigSelector = &mcSelector
+	infraMCP.Spec.MachineConfigSelector.MatchExpressions = []metav1.LabelSelectorRequirement{
+		metav1.LabelSelectorRequirement{
+			Key: "machineconfiguration.openshift.io/role",
+			Operator: metav1.LabelSelectorOpIn,
+			Values: []string{"worker", "infra"},
+		},
+	}
+	_, err = cs.MachineConfigPools().Create(infraMCP)
+	require.Nil(t, err)
+
+	infraMC := createMCToAddFileForRole("modify-host-file", "infra", "/etc/motd", "mco-test", "root")
```

Comment:
Since a bunch of other tests also write this, I think we could get false positives.  How about:
```
 createMCToAddFileForRole("modify-host-file", "infra", "/etc/mco-custom-pool", "custom-pool", "root")
```
or so?

(And we should in general have each test create a uniquely named file)

Reply:
oh yeah, makes total sense

Reply:
fixed, thanks!